src/server/serverinventorymgr.h

```diff
@@ -0,0 +1,54 @@
+/*
+Minetest
+Copyright (C) 2010-2020 Minetest core development team
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+
+#include "inventorymanager.h"
+#include <functional>
+
+class ServerEnvironment;
+
+class ServerInventoryMgr : public InventoryManager
+{
+public:
+	ServerInventoryMgr(ServerEnvironment *env);
+	virtual ~ServerInventoryMgr();
+	/*
+		Shall be called with the environment and the connection locked.
```

Comment:
What is "the connection" here?

Reply:
i don't know, it's just a code move, it's right that there is no interest to keep it there